### PR TITLE
Improve user search experience

### PR DESCRIPTION
**Pull request by ttannis**
### Summary of Changes

This PR further addresses issues from our user interviews, namely:
1. Persisting filters when re-executing a search on the `SearchPage`
2. Removing the requirement of a search term, which allows users to execute higher level searches across categories, e.g. "I'm looking for all resources in this schema"

**NavBar**
1. Passes the current location to `SearchBar`

**SearchBar**
1. Removed `isFormValid` check and requirement for a search term to execute a search
2. Passes a boolean to the `submitSearch` action to specify whether or not the existing filters should be used in this search.

**SearchPage**
1. Utilizes a new `hasFilters` boolean to assist with the error message logic. Our new user flow allows for users to clear all search input (search term + filters), in which case they now see a blank page. _Note: Will get feedback in Design QA about whether or not we should add some instruction, or otherwise better utilize this space._

**Redux Changes**
1. `submitSearch` and `searchAll` actions take a `useFilters` boolean in the payload.
2. `searchAll` no longer resets the filter state. Instead there is a new `clearAll` action that is fired by `searchAllWorker` if `useFilters` is false.

### Tests

TODO: As part of final feature branch review

### Documentation

TODO: As part of final feature branch review

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x)
- [x] PR includes a summary of changes, including screenshots of any UI changes. 
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [x] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

### CODE REVIEW

File: amundsen_application/static/js/components/common/SearchBar/index.tsx [hunk 1]
```diff
@@ -88,23 +90,16 @@ export class SearchBar extends React.Component<SearchBarProps, SearchBarState> {
 
   handleValueSubmit = (event: React.FormEvent<HTMLFormElement>) : void => {
     const searchTerm = this.state.searchTerm.trim();
+    const useFilters = this.props.location ? this.props.location.pathname === '/search' : false;
```
Comment (danwom) [inline amundsen_application/static/js/components/common/SearchBar/index.tsx]: Is there a better place for this logic? If possible I wouldn't want the `SearchBar` to be aware of `location` and which routes should preserve the filters. 

- Maybe the reducer or saga should be aware of the route and decide if the filters should be used or not.
- Maybe there should be a small checkbox somewhere that would explicitly decide to preserve or clear filters.

Comment (ttannis) [reply amundsen_application/static/js/components/common/SearchBar/index.tsx]: Hope the following is clear I time-boxed some investigation:

1. After some investigation I'm not confident this is a quick fix as opposed to separate task of augmenting our usage of Redux on that manner. I couldn't quite wrap my head around the safest way to do this that is 100% guaranteed to keep the redux state and react-router in sync -- if it exists.

2. If the ability to toggle `useFilters` or not was equal work effort I'd be confident to include it, but in this case there would be extra effort to add this to the application state and ensure page navigations still somehow update this state. I think it's worth investing effort into that toggle implementation only if users request it, or Design supports it. I'd essentially hate to spend a couple days on this and then have to remove it after Design QA. It is worth proposing as part of the final design QA though.

In summary we could talk about this more offline as its own task. I boil this down to the fact that we need a `SearchBar` that will cause either one of two action chains to happen:
1. clearFilters & submitSearch 
2. submitSearch

Meaning that `SearchBar` could take its submit method as a passed property, and it's parent components which are connected to the router could pass the correct submit action to the `SearchBar`. Let me know what you think, and I can add the modification before final feature branch review.

Comment (ttannis) [reply amundsen_application/static/js/components/common/SearchBar/index.tsx]: FYI references -- first found [this library](https://github.com/reactjs/react-router-redux) which might have been overcomplicating things and is now deprecated, but it could also be as little effort as some of[ these suggestions](https://www.pluralsight.com/guides/firing-redux-actions-on-route-transitions)
